src/core/unit.c

```diff
@@ -207,94 +206,129 @@ static void unit_init(Unit *u) {
                 UNIT_VTABLE(u)->init(u);
 }
 
+static int unit_add_another_name(Unit *u, char *name) {
```

Comment:
this takes possession of the name string allocation. That is not how we usually do things, hence should have a comment making this clear. Or maybe even rename the argument donated_name or so